library/runtime/src/nonJsMain/kotlin/io/matthewnelson/kmp/tor/runtime/util/PortUtil.kt

```diff
@@ -49,9 +107,26 @@ public expect fun Port.isAvailable(
  * @throws [IllegalArgumentException] if [limit] is not between 1 and 1_000 (inclusive)
  * @throws [IOException] if the check fails (e.g. calling from Main thread on Android)
  * */
-// Non-JS
 @Throws(IOException::class)
-public expect fun Port.Proxy.findAvailable(
+public fun Port.Proxy.findAvailable(
+    limit: Int,
+    host: LocalHost,
+): Port.Proxy = findAvailable(limit, host, null)
+
+@Throws(IOException::class, CancellationException::class)
+private fun Port.Proxy.findAvailable(
     limit: Int,
     host: LocalHost,
-): Port.Proxy
+    context: CoroutineContext?,
+): Port.Proxy {
+    val i = iterator(limit)
+    val ipAddress = host.resolve()
+    val inetAddress = ipAddress.toInetAddressWrapper()
+
+    while (context?.isActive != false && i.hasNext()) {
+        if (!inetAddress.isPortAvailable(i.next())) continue
```

Comment:
This could potentially throw something other than an `IOException` with how things are for native. Need to use a try/catch block here as well.